### ISSUE TITLE
Install unit testing tools and write `LocationSearch` unit tests

### ISSUE THREAD

**Issue by TylerMatteo** [NYCPlanning/ae-zoning-map-poc#60]
The goal of this Issue is to familiarize ourselves with the libraries we will use for front end unit testing in React apps. The two main libraries we want to use for this are:

* [Jest](https://jestjs.io/docs/getting-started) - A test runner that allows us to write structured tests with a robust `expect` API
* [React Testing Library (RTL)](https://testing-library.com/docs/) - Part of the `@testing-library` family of packages, RTL gives us a handle tool belt for writing user-centric unit tests. It is what we use inside of our Jest tests to do things like DOM manipulation and selection

There are two primary goals for this Issue.

- [ ] Install and configure Jest and RTL in `ae-zoning-map-poc`. See their [respective](https://jestjs.io/docs/getting-started) [docs](https://testing-library.com/docs/react-testing-library/intro) for instructions. Note that we should follow instructions for Typescript where given.
- [ ] Write a suite of unit tests for the `<LocationSearch>` component. This should test at least the following:
    - [ ] That the component correctly displays the list of boroughs from `useFindBoroughs` in the borough `<Select>`. This should require mocking that hook's return value
    - [ ] When a user selects a borough, block, and lot, then clicks the "Search" button, test that the `handleBblSearched` prop passed to the component is called with the correct arguments. This will likely involve spying on that prop and using Jest's [toHaveBeenCalledWith](https://jestjs.io/docs/expect#tohavebeencalledwitharg1-arg2-) method.

Part of this work will involve exploring how best to mock code that fetches data from an API, such the API call triggered by `useFindBoroughs`. We could simply mock return values for that hook based on what we expect it to return, but there may also be some tooling that can do that for us based on the API's OpenAPI specification. [Kubb](https://www.kubb.dev/) is a tool we already use in this repo to code-gen the hooks themselves based on the OpenAPI spec. Kubb also has a [plugin](https://www.kubb.dev/plugins/swagger-msw/) for [Mock Service Worker](https://mswjs.io/docs/) - a popular API mocking library. As part of this Issue, we should also explore using that tool chain for generating API mocks. If this work out, we can add these goals to the AC for this Issue

- [ ] Install the dependencies for MSW and MSW Kubb plugin and update `kubb.config.ts` to include it
- [ ] Regenerate the code in `/generated` to include the generated MSW code
- [ ] Use that newly generated code in the `LocationSearch` unit tests to mock the API call triggered by `useFindBoroughs`

Once all of the above is done, put a PR with these changes for review. You may also create a new Discussion about front end unit testing to document and get feedback on any issues you had, questions about general approach to unit testing, etc.

**Comment by TylerMatteo** [comment]
@bmarchena - @TangoYankee ran into some issues setting up Jest for the [repo template](https://github.com/NYCPlanning/ae-remix-template/issues/1) that led us to favor [Vitest](https://vitest.dev/). It's very similar to Jest but I think it would be worthwhile for us to switch over using it in the PoC repo as well. The feedback I left on your pull request about what to actually test should still be relevant. You can make a fresh branch to set up the repo with Vitest instead of Jest. Tim is available to help with with the installation and set up, and I suggest you two sync up today so Tim can show you how he did it in the repo template. Once you have the set up done, you can continue with writing tests per my feedback on your existing PR.

**Comment by bmarchena** [comment]
@TylerMatteo You mentioned that we'd be splitting the boroughs test into another issue this morning but could you clarify what you meant by "making the last test work?" I think that in order for us to test the submit functionality the way we'd like (ensuring the function is called with the certain params) using toHaveBeenCalledWith we'll need to resolve that other issue first but maybe you had another implementation or idea in mind

**Comment by TylerMatteo** [comment]
Hey @bmarchena, given the time we've sunk on this so far, I'd rather you focus on the data orchestration work or the Storybook testing over this. I'll self-assign and see if I can't refactor the component in a way that lets us write some tests now and revisit the issues we're having integrating MSW.

**Comment by TylerMatteo** [comment]
I'm having some trouble rebasing @bmarchena's branch for this but otherwise we can merge in that branch with the failing tests disabled for now. We should revisit how to take advantage of MSW in unit tests.
